core/parser/src/main/java/com/blazebit/persistence/parser/PathTargetResolvingExpressionVisitor.java

```diff
@@ -170,11 +171,26 @@ public PathTargetResolvingExpressionVisitor(EntityMetamodel metamodel, Type<?> s
 
     private Type<?> getType(Type<?> baseType, Attribute<?, ?> attribute) {
         Class<?> baseClass = baseType.getJavaType();
+        Member member = attribute.getJavaMember();
+
+        // Workaround HHH-12969: Hibernate returning the member of the identifier attribute of
+        // the IdClass as member of the association attribute on the entity.
+        if (attribute instanceof SingularAttribute) {
+            if (member instanceof Method) {
+                member = ReflectionUtils.getMethod(baseClass, member.getName());
+            } else if (member instanceof Field) {
+                member = ReflectionUtils.getField(baseClass, member.getName());
```

Comment:
Locally the test seemed to pass for Eclipselink, but maybe I messed something up in my profile configuration. I'll just wait for CI.

The code below makes me think that this code branch should work for Eclipselink too, but I could be mistaken. Do you want me to move this to the `JpaMetamodelAccessor` even if the existing tests would pass?

Reply:
Maybe it's easier if we use something like this

```
Class<?> clazz = com.blazebit.persistence.parser.util.JpaMetamodelUtils#resolveFieldClass(baseType.getJavaType(), attribute);
if (clazz != null) {
  return metamodel.type(clazz);
}

if (attribute instanceof PluralAttribute<?, ?, ?>) {
  return ((PluralAttribute<?, ?, ?>) attribute).getElementType();
}
return ((SingularAttribute<?, ?>) attribute).getType();
```

Reply:
I'm wondering whether its safe to assume that a field will exist with the attribute name, what if one uses method access rather than field access?

The current workaround looks at the member indicated by the attribute, and then looks up a similar typed member in the target class. Then for that member we resolve the actual type.

Now that I look into the code again, I do see though that the block has to be moved into the `if (baseClass != null) {` block.

Reply:
`com.blazebit.persistence.parser.util.JpaMetamodelUtils#resolveFieldClass` already does all the heavy lifting. The name is inappropriate now though since it also handles method members but the idea is that IMO the whole `getType` method can be reduced to the code I have shown.